plugin/documents.py

```diff
@@ -765,5 +787,12 @@ def clear_async() -> None:
 
         sublime.set_timeout_async(clear_async)
 
+    def _on_settings_object_changed(self) -> None:
+        new_syntax = self.view.settings().get("syntax")
+        if new_syntax != self._current_syntax:
+            self._current_syntax = new_syntax
+            self._cleanup()
+            self._setup()
+
```

Comment:
Don't we have to unregister the listener and then register it back and trigger "on_activated" to trigger didOpen?

Reply:
No need: container of SessionViews is cleared, _is_registered is set to False, and then on_activated_async is called by ST.

Reply:
I haven't checked but I've assumed that `on_activated_async` won't necessarily be called. Not unless the user changes the focus of the files manually.

Reply:
fixed in 5294c95